Blockzilla/BrowserViewController.swift

```diff
@@ -416,122 +426,142 @@ extension BrowserViewController: URLBarDelegate {
-
-            UIPasteboard.general.image = fetchImage(path)
+        
+        var url = URIFixup.getURL(entry: text)
+        if url == nil {
+            Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.typeQuery, object: TelemetryEventObject.searchBar)
+            url = searchEngineManager.activeEngine.urlForQuery(text)
+        } else {
+            Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.typeURL, object: TelemetryEventObject.searchBar)
         }
-
-        presentImageActionSheet(title: path, link: link, saveAction: saveAction, copyAction: copyAction)
+        if let overlayURL = url {
+            submit(url: overlayURL)
+            urlBar.url = overlayURL
+        }
+        urlBar.dismiss()
     }
+}
 
-    func browserDidStartNavigation(_ browser: Browser) {
+extension BrowserViewController: WebControllerDelegate {
+    func webControllerDidStartNavigation(_ controller: WebController) {
         urlBar.isLoading = true
         browserToolbar.isLoading = true
         urlBarContainer.isBright = false
         showToolbars()
     }
 
-    func browserDidFinishNavigation(_ browser: Browser) {
-        urlBar.url = browser.url
+    func webControllerDidFinishNavigation(_ controller: WebController) {
+        if webViewController.url?.absoluteString != "about:blank" {
+            urlBar.url = webViewController.url
+        }
         urlBar.isLoading = false
         browserToolbar.isLoading = false
         urlBarContainer.isBright = !urlBar.isEditing
         urlBar.progressBar.hideProgressBar()
     }
 
-    func browser(_ browser: Browser, didFailNavigationWithError error: Error) {
-        urlBar.url = browser.url
+    func webController(_ controller: WebController, didFailNavigationWithError error: Error) {
+        urlBar.url = webViewController.url
         urlBar.isLoading = false
         browserToolbar.isLoading = false
         urlBarContainer.isBright = true
     }
 
-    func browser(_ browser: Browser, didUpdateCanGoBack canGoBack: Bool) {
+    func webController(_ controller: WebController, didUpdateCanGoBack canGoBack: Bool) {
         urlBar.canGoBack = canGoBack
         browserToolbar.canGoBack = canGoBack
     }
 
-    func browser(_ browser: Browser, didUpdateCanGoForward canGoForward: Bool) {
+    func webController(_ controller: WebController, didUpdateCanGoForward canGoForward: Bool) {
         urlBar.canGoForward = canGoForward
         browserToolbar.canGoForward = canGoForward
     }
 
-    func browser(_ browser: Browser, didUpdateEstimatedProgress estimatedProgress: Float) {
+    func webController(_ controller: WebController, didUpdateEstimatedProgress estimatedProgress: Double) {
         // Don't update progress if the home view is visible. This prevents the centered URL bar
         // from catching the global progress events.
+        print("updating")
```

Comment:
remove print statements (unless we have a logger, but I think we can remove these?)